_posts/2025-05-08-opensearch-performance-3.0.md

```diff
@@ -0,0 +1,1013 @@
+
+![Force merge improvements](/assets/media/blog-images/2025-05-08-opensearch-performance-3.0/force-merge.png)
+
+These substantial gains in both storage efficiency and query performance demonstrate the value of using derived source for vectors when operating OpenSearch as a high-scale vector search engine. The transparent nature of this feature, which preserves critical functionality like reindexing and field-based updates, makes it available for wide range of users looking to optimize their vector search workloads in OpenSearch. For more information, see the [RFC](https://github.com/opensearch-project/k-NN/issues/2377).
+
+For benchmarking tests, see the [nightly benchmarks](https://benchmarks.opensearch.org/app/home). Benchmarks are run against a 10M/768D dataset.
+
+### Vector search performance boost up to 2.5x
+
+Concurrent segment search is [now enabled by default](https://github.com/opensearch-project/OpenSearch/pull/17978) for all vector search use cases. Concurrent segment search boosts vector query performance by up to **2.5x** with no impact on recall through parallelizing search queries across multiple threads. Additionally, changes to the floor segment size setting in the merge policy create more balanced segments and [improve tail latencies](https://github.com/opensearch-project/k-NN/pull/2623#issuecomment-2749056885) by up to **20%**. This performance boost is applicable to both in-memory and disk-optimized indexes.
+
+**Service time comparison**
+
+The following table compares 90th percentile (p90) service times, in milliseconds, across different k-NN engines with various concurrent segment search configurations.
+
+|k-NN engine	|Concurrent segment search disabled	|Concurrent segment search enabled (Lucene default number of slices)	|% Improvement	|Concurrent segment search with `max_slice_count` = 2	|% Improvement	|Concurrent segment search with `max_slice_count` = 4	|% Improvement	|Concurrent segment search with `max_slice_count` = 8	|% Improvement	|
+|---	|---	|---	|---	|---	|---	|---	|---	|---	|---	|
+|Lucene	|37	|15	|59.5	|16	|56.8	|15.9	|57	|16	|56.8	|
+|---	|---	|---	|---	|---	|---	|---	|---	|---	|---	|
+|NMSLIB	|35	|14	|60	|23	|34.3	|15	|57.1	|12	|65.7	|
+|Faiss	|37	|14	|62.2	|22	|40.5	|15	|59.5	|16	|56.8	|
+
+### Embedding processor optimization
+
+In OpenSearch 3.0, we added `skip_existing` support in the `text_embedding` processor to improve embedding ingestion performance. Previously, the processor made ML inference calls regardless of whether it was adding new embeddings or updating existing embeddings. With this feature, users now have an option to configure the processor so it skips inference calls when the target embedding field is not changed. This feature applies to both dense embeddings and sparse embeddings. By skipping the inference calls, this optimization can reduce ingestion latencies up to **70% for text embeddings**, **40% for text/image embeddings**, and **80% for sparse text embeddings**.
```

Comment:
```suggestion
In OpenSearch 3.0, we added `skip_existing` support in the `text_embedding` processor to improve embedding ingestion performance. Previously, the processor made ML inference calls regardless of whether it was adding new embeddings or updating existing embeddings. With this feature, users now have an option to configure the processor so that it skips inference calls when the target embedding field is not changed. This feature applies to both dense embeddings and sparse embeddings. By skipping the inference calls, this optimization can reduce ingestion latencies by up to **70% for text embeddings**, **40% for text/image embeddings**, and **80% for sparse text embeddings**.
```